subprojects/build-cache/src/main/java/org/gradle/caching/internal/origin/OriginMetadataFactory.java

```diff
@@ -76,7 +75,7 @@ public void execute(OutputStream outputStream) {
                 Properties properties = new Properties();
                 properties.setProperty(BUILD_INVOCATION_ID_KEY, currentBuildInvocationId.asString());
                 properties.setProperty(TYPE_KEY, entry.getClass().getCanonicalName());
```

Comment:
Why don't we use `getName()` instead of `getCanonicalName()`?

Reply:
No idea, but it makes sense to have a top-level class implement the executable thingies, so that we can look it up easily.

Reply:
+1 to the top level class. So your argument is that it doesn't make a difference if we use `getName` or `getCanonicalName`?

Reply:
Basically, they return the same, except `getCannonicalName()` returns `null` for non-top-level classes, which makes this code fail, which is a plus, so I'd leave it like this.

Reply:
I don't see why this code would fail - it would write `null` into the property and store an empty property in the `METADATA` properties file.
I think we should either make it fail or use `getName()`.